### PR TITLE
Style odata component

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Style odata component

**Related issue(s)**
https://github.com/kyma-project/console/issues/561

### CODE REVIEW

File: catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import AsyncApi from '@kyma-project/asyncapi-react';
+import {ODataReact} from '@kyma-project/odata-react';
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js]: `import { ODataReact } from '@kyma-project/odata-react';`

File: components/react-odata/src/ODataReact.tsx [hunk 2]
```diff
@@ -33,4 +36,4 @@ const ODataReact: React.FunctionComponent = () => {
   );
 };
 
-export default ODataReact;
+export { ODataReact };
```
Comment (magicmatatjahu) [inline components/react-odata/src/ODataReact.tsx]: Why not export default?

File: instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import AsyncApi from '@kyma-project/asyncapi-react';
+import {ODataReact} from '@kyma-project/odata-react';
```
Comment (magicmatatjahu) [inline instances/src/components/ServiceInstanceDetails/ServiceInstanceTabs/ServiceInstanceTabs.component.js]: `import { ODataReact } from '@kyma-project/odata-react';`

File: instances/src/commons/helpers.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ export const getDocumentsTypes = (svcClass, docsSortedByType = {}) => {
   let documentsTypes = [];
   if (svcClass) {
     if (docsSortedByType) documentsTypes = Object.keys(docsSortedByType);
-    if (svcClass.apiSpec) documentsTypes.push('Console');
+    if (svcClass.openApiSpec) documentsTypes.push('Console');
```
Comment (magicmatatjahu) [inline instances/src/commons/helpers.js]: Where is condition for `odataSpec`? This is unnecessary?

File: components/react-odata/src/tools/V2V3toV4.ts [hunk 5]
```diff
@@ -1140,3 +1140,4 @@ export default `<?xml version="1.0" encoding="utf-8"?>
   </xsl:template>
 
 </xsl:stylesheet>`;
+export { V2V3toV4 };
```
Comment (magicmatatjahu) [inline components/react-odata/src/tools/V2V3toV4.ts]: Why not export default?

File: components/react-odata/src/components/styled/styled.ts [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+export const CollapseButton = styled(Button)`
+  && {
+    direction: rtl;
+    margin-bottom: 20px;
+  }
+  &:before {
+    margin-right: 0;
+    margin-left: 8px;
+    transition: 0.3s ease;
+    ${(props: { open?: boolean }) => props.open && "transform: rotate(90deg);"};
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/styled/styled.ts]: Please create Interface for this styled component and then use this Interface here. `(props: Interface) =>`

File: components/react-odata/src/components/styled/styled.ts [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+export const CollapseArrow = styled(Icon)`
+  && {
+    margin-left: 5px;
+    position: relative;
+    display: inline-block;
+    cursor: pointer;
+  }
+  &:before {
+    transition: 0.3s ease;
+    ${(props: { open?: boolean }) => props.open && "transform: rotate(90deg);"};
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/styled/styled.ts]: Please create Interface for this styled component and then use this Interface here. `(props: Interface) =>`

File: catalog/src/commons/helpers.js [hunk 8]
```diff
@@ -29,7 +29,7 @@ export const getDocumentsTypes = (items, docsSortedByType) => {
   let documentsTypes = [];
   if (items) {
     if (docsSortedByType) documentsTypes = Object.keys(docsSortedByType);
-    if (items.apiSpec) documentsTypes.push('Console');
+    if (items.openApiSpec) documentsTypes.push('Console');
```
Comment (magicmatatjahu) [inline catalog/src/commons/helpers.js]: Where is condition for `odataSpec`? This is unnecessary?

File: components/react-odata/src/components/Table/MainDataTable/CollapsibleRow.tsx [hunk 9]
```diff
@@ -14,26 +15,26 @@ const CollapsibleRow: React.FunctionComponent<Props> = ({
...
-        </tr>
+          </TableCell>
+        </TableRow>
       )}
     </>
   );
 };
 
-export default CollapsibleRow;
+export { CollapsibleRow };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/MainDataTable/CollapsibleRow.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/MainDataTable/HideableSubTable.tsx [hunk 10]
```diff
@@ -17,27 +24,27 @@ const HideableSubTable: React.FunctionComponent<Props> = ({ data }) => {
...
+          </TableRow>
         ))}
-      </tbody>
+      </TableBody>
     </StyledTable>
   );
 };
 
-export default HideableSubTable;
+export { HideableSubTable };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/MainDataTable/HideableSubTable.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/MainDataTable/Table.tsx [hunk 11]
```diff
@@ -28,39 +46,64 @@ const Table: React.FunctionComponent<Props> = ({
...
         )}
-      </tbody>
-    </table>
+      </TablePanel>
+    </TableWrapper>
   );
 };
 
-export default Table;
+export { Table };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/MainDataTable/Table.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/MainDataTable/Table.tsx [hunk 12]
```diff
@@ -28,39 +46,64 @@ const Table: React.FunctionComponent<Props> = ({
...
+    const ref = useRef<boolean>();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
+
+  const [show, setShow] = useState<boolean>(showAll);
+  const prevShowAll = usePrevious(showAll);
+  useEffect(() => {
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/MainDataTable/Table.tsx]: Please add blank lines after and before `useState`, `useEffect` etc. for better code readability.

File: components/react-odata/src/components/Table/ServiceDocumentationTable/CollapsibleAnnotation.tsx [hunk 13]
```diff
@@ -15,31 +23,31 @@ const CollapsibleAnnotation: React.FunctionComponent<Props> = ({ data }) => {
...
         )}
-      </tbody>
-    </table>
+      </TableBody>
+    </StyledTable>
   );
 };
 
-export default CollapsibleAnnotation;
+export { CollapsibleAnnotation };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/CollapsibleAnnotation.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/ServiceDocumentationTable/CollapsibleTable.tsx [hunk 14]
```diff
@@ -24,59 +32,60 @@ const CollapisbleTable: React.FunctionComponent<Props> = ({ data }) => {
...
           );
         })}
-      </tbody>
+      </TableBody>
     </StyledTable>
   );
 };
 
-export default CollapisbleTable;
+export { CollapsibleTable };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/CollapsibleTable.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx [hunk 15]
```diff
@@ -1,46 +1,122 @@
...
+    const ref = useRef<boolean>();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  }
+
+  const [show, setShow] = useState<boolean>(showAll);
+  const prevShowAll = usePrevious(showAll);
+  useEffect(() => {
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx]: Please add blank lines after and before `useState`, `useEffect` etc. for better code readability.

File: components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx [hunk 16]
```diff
@@ -1,46 +1,122 @@
...
+            </TableBody>
+          </StyledTable>
+        )}
+      </TablePanel>
+    </TableWrapper>
   );
+};
 
-export default ServiceDocumentationTable;
+export { ServiceDocumentationTable };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/ServiceDocumentationTable/SimpleTable.tsx [hunk 17]
```diff
@@ -1,24 +1,34 @@
...
+        </TableRow>
       ))}
-    </tbody>
-  </table>
+    </TableBody>
+  </StyledTable>
 );
 
-export default SimpleTable;
+export { SimpleTable };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/SimpleTable.tsx]: Please use `export default`

File: components/react-odata/src/components/Table/TableContainer.tsx [hunk 18]
```diff
@@ -75,4 +83,4 @@ const TableContainer: React.FunctionComponent<Props> = ({ arg }) => {
   );
 };
 
-export default TableContainer;
+export { TableContainer };
```
Comment (magicmatatjahu) [inline components/react-odata/src/components/Table/TableContainer.tsx]: Please use `export default``

File: catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js (hunk 1 above)
Comment (pkosiec) [reply catalog/src/components/ServiceClassDetails/ServiceClassTabs/ServiceClassTabs.component.js]: It should be formatted automatically. Please run `npm run bootstrap` to install pre-commit hook.

File: components/react-odata/src/ODataReact.tsx (hunk 2 above)
Comment (pkosiec) [reply components/react-odata/src/ODataReact.tsx]: It's easier to use - imports in IDE are autocompleted

File: components/react-odata/src/components/ErrorComponent/ErrorComponent.tsx [hunk 19]
```diff
@@ -1,20 +1,32 @@
...
   const data = error.children.map((elem: Node) => elem.value);
   return (
-    <div>
-      <h3>{data[0]}</h3>
-      <code>{data[1]}</code>
-      <p>{data[2]}</p>
-    </div>
+    <PageWrapper>
+      <Alert dismissible={true} type="error">
+        <h3>{data[0]}</h3>
```
Comment (pkosiec) [inline components/react-odata/src/components/ErrorComponent/ErrorComponent.tsx]: Please declare:
```
const [header, code, p] = data;
```
and use `header`, `code`, and `p` in code

File: components/react-odata/src/components/Table/MainDataTable/CollapsibleRow.tsx (hunk 9 above)
Comment (pkosiec) [reply components/react-odata/src/components/Table/MainDataTable/CollapsibleRow.tsx]: Or at least use `export const CollapsibleRow in first line of the component`

File: components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx [hunk 20]
```diff
@@ -1,46 +1,122 @@
...
+  useEffect(() => {
+    if (prevShowAll !== undefined && prevShowAll !== showAll) {
+      setShow(showAll);
+    }
+  }, [prevShowAll, showAll]);
+  const [showPart, setShowPart] = useState<boolean[]>(
+    Array(data.length).fill(false),
+  );
+
+  return !Array.isArray(data) ? null : (
```
Comment (pkosiec) [inline components/react-odata/src/components/Table/ServiceDocumentationTable/ServiceDocumentationTable.tsx]:
```
if (!Array.isArray(data)) {
  return null;
}

return <TableWrapper...
```

File: components/react-odata/tslint.json [hunk 21]
```diff
@@ -21,10 +23,19 @@
...
+    "prefer-object-spread": true,
+    "no-boolean-literal-compare": true
   },
   "linterOptions": {
-    "exclude": ["src/serviceWorker.ts", "node_modules/**/*.ts"]
+    "exclude": [
+      "src/serviceWorker.ts",
+      "**/*.json",
+      "node_modules/**/*.ts",
+      "src/ODataFiles/*.ts", // remove this later
```
Comment (pkosiec) [inline components/react-odata/tslint.json]: Remove comment from JSON file

**Pull request merged**